frontend/src/main/components/Courses/CoursesForm.js

```diff
@@ -0,0 +1,156 @@
+import { Button, Form} from 'react-bootstrap';
+import { useForm } from 'react-hook-form'
+import { useNavigate } from 'react-router-dom'
+
+function CoursesForm({ initialContents, submitAction, buttonLabel = "Create" }) {
+
+    // Stryker disable all
+    const {
+        register,
+        formState: { errors },
+        handleSubmit,
+    } = useForm(
+        { defaultValues: initialContents || {}, }
+    );
+    // Stryker restore all
+
+    const navigate = useNavigate();
+
+    const testIdPrefix = "CoursesForm";
+
+    // For explanation, see: https://stackoverflow.com/questions/3143070/javascript-regex-iso-datetime
+    // Note that even this complex regex may still need some tweaks
+
+    // Stryker disable next-line Regex
```

Comment:
When using `type="datetime-local"` it might not be necessary to also use a regular expression to verify the datetime.   That's typically only necessary when the field is being entered in plain text.   In this case, I think the react component is already ensuring that invalid data can't be entered.  

We might be able to simplify the code by removing this.